app/controllers/users/sessions_controller.rb

```diff
@@ -204,7 +204,10 @@ def handle_valid_authentication
     def track_authentication_attempt
       user = user_from_params || AnonymousUser.new
 
-      success = current_user.present? && !user_locked_out?(user) && valid_captcha_result?
+      success = current_user.present? &&
+                !user_locked_out?(user) &&
+                (valid_captcha_result? || log_captcha_failures_only?)
```

Comment:
Was curious if we could bake the log-only consideration into `valid_captcha_result?`, but I see that it's valuable to include the raw result in the logging value below for `valid_captcha_result`. Nice separation 👍